BraveShared/Analytics/UserReferralProgram.swift

```diff
@@ -50,7 +50,9 @@ public class UserReferralProgram {
             
             if ref.isExtendedUrp() {
                 if let headers = ref.customHeaders {
-                    Preferences.URP.customHeaderData.value = NSKeyedArchiver.archivedData(withRootObject: headers)
+                    do {
+                        try Preferences.URP.customHeaderData.value = NSKeyedArchiver.archivedData(withRootObject: headers, requiringSecureCoding: false)
+                    } catch {}
```

Comment:
Can you add an error here?

```suggestion
                    } catch {
                        log.error("Failed to save URP custom header data \(headers) with error: \(error.localizedDescription)")
                    }
```

Reply:
Will do.